From 71e0b434a9c9eaa4879cd71e203b02dbfde28553 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Sun, 23 Feb 2025 18:02:34 +0100
Subject: [PATCH 1/4] Fix "yarock-player.org" organisation breaking appid,
 de-duplicate desktop files

This (resulting in "org.yarock-player.yarock") did not match both
org.yarock.desktop and yarock.desktop, resulting in the wayland
placeholder icon used instead of Yarock's, among other potential
XDG/DBus adjacent issues.

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt          |  2 +-
 src/main.cpp            |  4 --
 2 files changed, 1 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6dd3952..69654c6 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -203,7 +203,7 @@ install(TARGETS yarock DESTINATION ${CMAKE_INSTALL_BIN})
 install(FILES ${YAROCK_QM_FILES} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/yarock/translations")
 
 # -- install destop file
-install(FILES data/org.yarock.desktop  DESTINATION "${CMAKE_INSTALL_PREFIX}/share/applications")
+install(FILES data/yarock.desktop DESTINATION "${CMAKE_INSTALL_PREFIX}/share/applications")
 
 # -- install appdata file
 install(FILES data/org.yarock.metainfo.xml  DESTINATION "${CMAKE_INSTALL_PREFIX}/share/metainfo")
diff --git a/src/main.cpp b/src/main.cpp
index 03eeafa..276d8de 100755
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -24,8 +24,6 @@ int main(int argc, char *argv[]) {
     //! core application settings
     QCoreApplication::setApplicationName(APP_NAME);
     QCoreApplication::setApplicationVersion(VERSION);
-    QCoreApplication::setOrganizationName(ORG_NAME);
-    QCoreApplication::setOrganizationDomain("yarock-player.org");
 
     //! taken from amarok
     // This call is needed to prevent a crash on exit with Phonon-VLC and LibPulse
@@ -40,8 +38,6 @@ int main(int argc, char *argv[]) {
     QApplication application(argc, argv);
     application.setApplicationName(APP_NAME);
     application.setApplicationVersion(VERSION);
-    application.setOrganizationName(ORG_NAME);
-    application.setOrganizationDomain("yarock-player.org");
     application.setAttribute(Qt::AA_DontShowIconsInMenus, false);
 
     //! Qt Type registration
-- 
2.48.1


From d0b1157c61cfe520d03b656f35ac3c914dbb9f04 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Sun, 23 Feb 2025 18:09:11 +0100
Subject: [PATCH 2/4] Use organisationless metainfo.xml file

Follow-up to 71e0b434a9c9eaa4879cd71e203b02dbfde28553, incidentally
running dos2unix over it.

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt                                |   2 +-
 ...arock.metainfo.xml => yarock.metainfo.xml} | 102 +++++++++---------
 2 files changed, 52 insertions(+), 52 deletions(-)
 rename data/{org.yarock.metainfo.xml => yarock.metainfo.xml} (98%)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 69654c6..995c337 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -206,7 +206,7 @@ install(FILES ${YAROCK_QM_FILES} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/yaro
 install(FILES data/yarock.desktop DESTINATION "${CMAKE_INSTALL_PREFIX}/share/applications")
 
 # -- install appdata file
-install(FILES data/org.yarock.metainfo.xml  DESTINATION "${CMAKE_INSTALL_PREFIX}/share/metainfo")
+install(FILES data/yarock.metainfo.xml  DESTINATION "${CMAKE_INSTALL_PREFIX}/share/metainfo")
 
 # -- install icons files
 install(FILES ${CMAKE_SOURCE_DIR}/icon/yarock_16x16.png DESTINATION "${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/16x16/apps" RENAME application-x-yarock.png)
diff --git a/data/org.yarock.metainfo.xml b/data/yarock.metainfo.xml
similarity index 98%
rename from data/org.yarock.metainfo.xml
rename to data/yarock.metainfo.xml
index 224713a..3afa371 100755
-- 
2.48.1
